### ISSUE TITLE
Autodiscover Exchange gets wrong server

### ISSUE THREAD

**Issue by ggmanugg** [mailcow/mailcow-dockerized#2947]
Hi,

I finished the setup of the mailcow server with no errors. The receipt and the sending of mails works without any problems.

Now I wanted to add my mail configs to my android phone. But now I have the following problem:

When i add the mail account to gmail (exchange) I have to manually add the mailserver. The autoconfig just add example.com instead of mail.example.com
After i edited it to mail.example.com it works and can connect to it.

Does anybody knows what the problem is? Thanks in advance

**Comment by andryyy** [comment]
Hi, is autodiscover correctly setup, too?

**Comment by ggmanugg** [comment]
I don't changed anything on the php.ini config

But I have an other idea, maybe it's because i changed my hostname and the php config has not checked it yet.
Does mailcow reload the php settings after every reload of the docker containers?

**Comment by andryyy** [comment]
There is no php.ini to be changed. :) What exactly did you change? Does "autodiscover.domainname.tld" exist in DNS?

**Comment by ggmanugg** [comment]
Yes but the php.ini needs the mailcow_hostname right? Before i started the containers with the hostname example.com and now have it with mail.example.com.

Yes, all DNS records are setup up correctly.

**Comment by ggmanugg** [comment]
It seems to be a problem by Gmail/Microsoft.
When  i type the e-mail address like testuser@mail.example.com it adds mail.example.com to the server field. 
@andryyy Do you don't have the same problem?

Have always to manually correct this entry. Afterwards i can add my mail-address. 

Btw. The autodiscovery test by microsoft have been passed everytime.

**Comment by ggmanugg** [comment]
Another addition:

This problem just happens when the autodiscover.xml is detected by dns entries. I run my Mailcow-Server behind a sophos firewall where I just allowed to proxy mail.example.com. After I added autodiscover.example.com and autoconfig.example.com to the list it detects the autodiscover over the Autodiscover URL. Then i had not to modify the server field because it doen'st showed up.

Thanks anyways for your time.
